src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts

```diff
@@ -166,12 +159,33 @@ class editDocument extends viewModelBase {
         this.addNotification(this.changeNotification);
     }
 
-    private initializeObservables(): void {
-        this.userSpecifiedIdCustomValidityError = ko.computed(() => {
-            const documentId = this.userSpecifiedId();
-            return documentId.includes("\\") ? "Document name cannot contain '\\'" : "";
+    private initValidation() {      
+        const rg1 = /^[^\\]*$/; // forbidden character - backslash
+        this.userSpecifiedId.extend({           
+            validation: [
+                {
+                    validator: (val: string) => rg1.test(val),
+                    message: "Can't use backslash in document name"
+                }]
         });
 
+        this.documentText.extend({
+            required: true
+        });
+    }
+
+    private initializeObservables(): void {
+        
+        this.dirtyFlag = new ko.DirtyFlag([this.documentText, this.userSpecifiedId], false, jsonUtil.newLineNormalizingHashFunction); 
+          
+        this.isSaveEnabled = ko.pureComputed(() => {            
+            if (this.isSaving() ||
+                (!this.dirtyFlag().isDirty() && this.metadata().etag()) ) {
```

Comment:
assign is dirty to variable, then use it to properly register dependencies of pure computed.